### PR TITLE
BDD  TA Frameworks 

### PR DESCRIPTION
**Pull request by KseniiaOvsiannikova**
Home Work M11

### CODE REVIEW

File: src/main/java/pages/AbstractPage.java [hunk 1]
```diff
@@ -36,5 +40,10 @@ protected boolean waitForElementIsPresent(WebElement element) {
 
         return element.isEnabled();
     }
+
+    protected List<WebElement> findElements(By by) {
```
Comment (olelysieev-clgx) [inline src/main/java/pages/AbstractPage.java]: please remove unused method

File: src/main/java/pages/cucumber/CucumberToolsPage.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    public CucumberToolsPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public boolean isPageWithTitleDisplayed() {
+        return actualPageTitle.isDisplayed();
+    }
+
+    @Override
+    protected AbstractPage openPage() {
```
Comment (olelysieev-clgx) [inline src/main/java/pages/cucumber/CucumberToolsPage.java]:
```
protected CucumberToolsPage openPage() {
 return new CucumberToolsPage(driver);
}
```

File: src/test/java/bdd/runner/CucumberTestRunner.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package bdd.runner;
+
+import io.cucumber.testng.AbstractTestNGCucumberTests;
+import io.cucumber.testng.CucumberOptions;
+
+@CucumberOptions(
+        plugin = "pretty",
+        strict = true,
+        monochrome = true,
+        tags = "@smoke",
```
Comment (olelysieev-clgx) [inline src/test/java/bdd/runner/CucumberTestRunner.java]: remove this tag

File: src/test/java/bdd/steps/StepDefinitions.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+public class StepDefinitions {
+
+    @When("the user opens Cucumber website")
+    public void openCucumberWebsite() {
+        new CucumberHomePage(DriverManager.getDriver()).openPage();
+    }
+
+    @When("the user clicks on the {string} section")
+    public void clickOnSection() {
```
Comment (olelysieev-clgx) [inline src/test/java/bdd/steps/StepDefinitions.java]: ...(String section)...

File: src/test/java/bdd/steps/StepDefinitions.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        new CucumberHomePage(DriverManager.getDriver()).menuDropdownSection.click();
+    }
+
+    @When("the user clicks on the {string} item")
+    public void clickOnItem() {
```
Comment (olelysieev-clgx) [inline src/test/java/bdd/steps/StepDefinitions.java]: ...(String item)...

File: src/main/java/pages/AbstractPage.java (hunk 1 above)
Comment (KseniiaOvsiannikova) [reply src/main/java/pages/AbstractPage.java]: Deleted, thank you for the pointing.
Please see in the following commit.

File: src/main/java/pages/cucumber/CucumberToolsPage.java (hunk 2 above)
Comment (KseniiaOvsiannikova) [reply src/main/java/pages/cucumber/CucumberToolsPage.java]: Refactored, thank you.
Please see in the following commit.

File: src/test/java/bdd/runner/CucumberTestRunner.java (hunk 3 above)
Comment (KseniiaOvsiannikova) [reply src/test/java/bdd/runner/CucumberTestRunner.java]: Refactored, thank you.
Please see in the following commit.

File: src/test/java/bdd/steps/StepDefinitions.java (hunk 4 above)
Comment (KseniiaOvsiannikova) [reply src/test/java/bdd/steps/StepDefinitions.java]: Refactored, thank you.
Please see in the following commit.

File: src/test/java/bdd/steps/StepDefinitions.java (hunk 5 above)
Comment (KseniiaOvsiannikova) [reply src/test/java/bdd/steps/StepDefinitions.java]: Refactored, thank you.
Please see in the following commit.

File: pom.xml [hunk 6]
```diff
@@ -107,6 +107,22 @@
...
+            <artifactId>cucumber-java</artifactId>
+            <version>4.8.1</version>
+        </dependency>
+        <dependency>
+            <groupId>io.cucumber</groupId>
+            <artifactId>cucumber-testng</artifactId>
+            <version>4.8.1</version>
+        </dependency>
+        <dependency>
+            <groupId>io.qameta.allure</groupId>
```
Comment (olelysieev-clgx) [inline pom.xml]: we are using yandex allure, not qameta

File: src/main/java/pages/AbstractPage.java [hunk 7]
```diff
@@ -36,5 +36,6 @@ protected boolean waitForElementIsPresent(WebElement element) {
 
         return element.isEnabled();
     }
+
```
Comment (olelysieev-clgx) [inline src/main/java/pages/AbstractPage.java]: empty line

File: src/test/java/bdd/runner/CucumberTestRunner.java [hunk 8]
```diff
@@ -0,0 +1,17 @@
+package bdd.runner;
+
+import io.cucumber.testng.AbstractTestNGCucumberTests;
+import io.cucumber.testng.CucumberOptions;
+
+@CucumberOptions(
+        plugin = {"pretty", "html:target/cucumber-reports",
+                "json:target/cucumber-reports/CucumberTests.json",
+                "testng:target/cucumber-reports/CucumberTests.xml"},
+                //"io.qameta.allure.cucumberjvm.AllureCucumberJvm"},
```
Comment (olelysieev-clgx) [inline src/test/java/bdd/runner/CucumberTestRunner.java]: commented code

File: src/test/java/bdd/steps/StepDefinitions.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+import pages.cucumber.CucumberHomePage;
+
+public class StepDefinitions {
+
+    @After
+    public void closeBrowser() {
+        DriverManager.closeDriver();
+    }
+
+    @When("the user opens Cucumber website")
```
Comment (olelysieev-clgx) [inline src/test/java/bdd/steps/StepDefinitions.java]: @Given, not @When

File: src/test/resources/features/MenuSections.feature [hunk 10]
```diff
@@ -0,0 +1,12 @@
+Feature: As a user I want to get ability to use menu sections
+
+  Scenario Outline: Tools sections should contain appropriate columns
+    Given the user opens Cucumber website
```
Comment (olelysieev-clgx) [inline src/test/resources/features/MenuSections.feature]: This line can be used as Background

File: src/main/java/pages/AbstractPage.java (hunk 7 above)
Comment (KseniiaOvsiannikova) [reply src/main/java/pages/AbstractPage.java]: Deleted, thank you for pointing.
Please see in the following commit.

File: src/test/java/bdd/steps/StepDefinitions.java (hunk 9 above)
Comment (KseniiaOvsiannikova) [reply src/test/java/bdd/steps/StepDefinitions.java]: Changed, thank you for pointing.
Please see in the following commit.

File: src/test/resources/features/MenuSections.feature (hunk 10 above)
Comment (KseniiaOvsiannikova) [reply src/test/resources/features/MenuSections.feature]: Refactored, thank you for pointing.
Please see in the following commit.

File: pom.xml (hunk 6 above)
Comment (KseniiaOvsiannikova) [reply pom.xml]: Deleted, thank you.
Please see in the following commit.

File: src/test/java/bdd/runner/CucumberTestRunner.java (hunk 8 above)
Comment (KseniiaOvsiannikova) [reply src/test/java/bdd/runner/CucumberTestRunner.java]: Deleted, thank you for pointing.
Please see in the following commit.

File: src/test/java/bdd/runner/CucumberTestRunner.java [hunk 11]
```diff
@@ -0,0 +1,18 @@
+package bdd.runner;
+
+import io.cucumber.testng.AbstractTestNGCucumberTests;
+import io.cucumber.testng.CucumberOptions;
+
+@CucumberOptions(
+        plugin = {"pretty", "html:target/cucumber-reports",
+                "json:target/cucumber-reports/CucumberTests.json",
+                "testng:target/cucumber-reports/CucumberTests.xml",
+                "json:target/Cucumber.json",
```
Comment (olelysieev-clgx) [inline src/test/java/bdd/runner/CucumberTestRunner.java]: why you need lines 10-11 ? Duplication of lines 7 and 8

**Review by olelysieev-clgx — APPROVED** [review]

File: src/test/java/bdd/runner/CucumberTestRunner.java (hunk 11 above)
Comment (KseniiaOvsiannikova) [reply src/test/java/bdd/runner/CucumberTestRunner.java]: Deleted, thank you for pointing.
Please see in the following commit.

**Pull request merged**